core/server/data/migrations/versions/4.16/01-add-custom-theme-settings-table.js

```diff
@@ -0,0 +1,12 @@
+const utils = require('../../utils');
+
+module.exports = utils.addTable('custom_theme_settings', {
+    id: {type: 'string', maxlength: 24, nullable: false, primary: true},
+    theme: {type: 'string', maxlength: 191, nullable: false},
+    property: {type: 'string', maxlength: 50, nullable: false},
+    type: {type: 'string', maxlength: 50, nullable: false},
+    group: {type: 'string', maxlength: 50, nullable: true},
+    options: {type: 'text', maxlength: 65535, nullable: true}, // json
+    default_value: {type: 'text', maxlength: 65535, nullable: false},
```

Comment:
This data isn't strictly necessary to store because the source of truth will always be the theme and we'll extract that information when validating the theme at startup/activation. However, there are potentially some cases where it would be nice to have the data (eg, informing the user that new options have been added when upgrading a theme) so I've added kept them here for now.

Reply:
In the spirit of Hyrem's law, if we don't have a usecase for it, I think it would be better to not add more columns?

Reply:
👍 Removed them for now, they were complicating the syncing code unnecessarily too.